internal/provisioner/kops_provisioner.go

```diff
@@ -187,14 +187,27 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster) error
 		return err
 	}
 
-	// TODO: Rework this as we make the API calls asynchronous.
-	wait = 60
-	logger.Infof("waiting up to %d seconds for mattermost operator to start...", wait)
-	pod, err := k8sClient.WaitForPodRunning("mattermost-operator", "mattermost-operator", wait)
-	if err != nil {
-		return err
+	operators := []string{"mysql-operator", "mattermost-operator"}
+	for _, operator := range operators {
+		pods, err := k8sClient.GetPodsFromDeployment(operator, operator)
+		if err != nil {
+			return err
+		}
+		if len(pods.Items) == 0 {
+			return fmt.Errorf("no pods found from %q deployment", operator)
```

Comment:
Are we guaranteed to find pods this early? Or might the pods themselves be slow to start?

Reply:
This is actually something I am unsure of. I never ran into an issue, but it is definitely a concern that I will look into further.

Very shortly after the deployment is created (if not immediately) the pod objects will be created. This will happen far before they ever get scheduled on the cluster. That said, there may be a small period of time where the deployment exists, but no pods. I will investigate further, but propose that we move forward with this for now.

Reply:
I'm good with using this for now. Might be worth having a chat with @cpanato too about best ways to check the ready status of operators